core/vm/vmcontext/caller.go

```diff
@@ -0,0 +1,84 @@
+package vmcontext
+
+import (
+	"math/big"
+
+	"github.com/celo-org/celo-blockchain/common"
+	"github.com/celo-org/celo-blockchain/core/types"
+	"github.com/celo-org/celo-blockchain/core/vm"
+	"github.com/celo-org/celo-blockchain/params"
+)
+
+type evmCaller struct {
```

Comment:
I feel like the evmCaller and its factory would be better placed in the contracts package, it doesn't seem to have much to do with vmcontext

Reply:
I also wonder if this would fix the cyclic dependency issue when contracts are migrated to use this approach.

Reply:
Which cyclic dependency is this?

Reply:
not best pkg name for sure.

The problem is that

* `Context` depends on Contracts

Reply:
I spent some time trying to refactor this, but it's quite involved, so probably best left to another PR